### PR TITLE
feat(BBD-543): Infinite-scroll

### PR DESCRIPTION
**Pull request by Vic-Dev**
Switch to using currentRecordCount w/ pastpurchase rather than allRecordCount

### CODE REVIEW

File: src/core/reducers/ui.ts [hunk 1]
```diff
@@ -18,12 +18,16 @@ export default function updateUi(state: State = {}, { type, payload, meta = <Act
...
 
-export const createComponentState = (state: State, { tagName, id, state: tagState }: Actions.Payload.Component.State) =>
+// tslint:disable-next-line max-line-length
+export const createComponentState = (state: State, { tagName, id, state: tagState, persist }: Actions.Payload.Component.State) =>
   ({
     ...state,
     [tagName]: {
       ...state[tagName],
-      [id]: tagState
+      [id]: {
```
Comment (treeforever) [inline src/core/reducers/ui.ts]: why the change?

File: src/core/requests.ts [hunk 2]
```diff
@@ -39,13 +39,14 @@ namespace Requests {
     return <Request>Requests.chain(config.search.defaults, request, config.search.overrides);
   };
 
-  export const pastPurchaseProducts = (state: Store.State, getNavigations: boolean = false): Request => {
+  // tslint:disable-next-line max-line-length
+  export const pastPurchaseProducts = (state: Store.State, getNavigations: boolean = false, options: { pageSize: number, skip: number} = { pageSize: Selectors.pastPurchasePageSize(state), skip: Selectors.pastPurchasePageSize(state) * (Selectors.pastPurchasePage(state) - 1) }): Request => {
```
Comment (treeforever) [inline src/core/requests.ts]: Consider to pass in this long list of parameters as an object or get default value in the body of the function. It causes difficulty to read when it gets long.

Comment (treeforever) [reply src/core/requests.ts]: One question: when you pass in options, what if `pageSize` is defined and `skip` is undefined, it won't get the default value I assume? Or maybe this scenario won't occur?

**Comment by treeforever** [comment]
I also have a general question. Since we've seen some clients with massive data for products blowing up firefox and IE, it may be worth considering how infinite scroll can hold so much products data in the state.

File: src/core/reducers/ui.ts (hunk 1 above)
Comment (Vic-Dev) [reply src/core/reducers/ui.ts]: Wanted to add a "persist" flag so that info could persist over time

File: src/core/requests.ts (hunk 2 above)
Comment (Vic-Dev) [reply src/core/requests.ts]: You're right, it won't work. But! If you use this syntax, it does:
`{ pageSize = Selectors.pastPurchasePageSize(state), skip = Selectors.pastPurchasePageSize(state) * (Selectors.pastPurchasePage(state) - 1) }: { pageSize?: number, skip?: number}`

I switched it to not passing it in as default though, because you're right that line's getting too long. :) Just thought that syntax was kewl.

File: src/core/store/index.ts [hunk 3]
```diff
@@ -90,7 +92,10 @@ namespace Store {
   export interface UI {
     [tagName: string]: {
       global?: any;
-      [tagId: string]: any;
+      [tagId: number]: {
```
[the remark this answers is not in the document]
Comment (Vic-Dev) [reply src/core/store/index.ts]: This was going to be for infinite isFetchingForward & isFetchingBackward state, but I ended up creating a section in the store for InfiniteScroll.

I can take it out if you think I should.

[the remark this answers is not in the document]
Comment (treeforever) [reply src/core/store/index.ts]: Ok I see. If it is not used then we should take it out I think :)
